### PR TITLE
disttask: refine the error handling interface

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Issue Number: ref #41495

Problem Summary:

### What is changed and how it works?

This PR changes the type of error from `[]byte` to `error`, which is more intuitive. For serialization/deserialization, I use `pingcap/errors.MarshalJSON` and `pingcap/errors.UnmarshalJSON`.

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by ywqzzy — APPROVED** [review]

**Review by wjhuang2016 — APPROVED** [review]

File: disttask/framework/storage/task_table.go [hunk 1]
```diff
@@ -107,7 +107,16 @@ func row2GlobeTask(r chunk.Row) *proto.Task {
 		Meta:         r.GetBytes(7),
 		Concurrency:  uint64(r.GetInt64(8)),
 		Step:         r.GetInt64(9),
-		Error:        r.GetBytes(10),
+	}
+	if !r.IsNull(10) {
+		errBytes := r.GetBytes(10)
+		stdErr := errors.Normalize("")
+		err := stdErr.UnmarshalJSON(errBytes)
+		if err != nil {
```
Comment (zimulala) [inline disttask/framework/storage/task_table.go:115]: Do we need add an error to `task.Error`? Otherwise this error be omitted.

**Review by zimulala — APPROVED** [review]
LGTM

**Pull request merged**